recipes/mingw/all/conanfile.py

```diff
@@ -0,0 +1,54 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class MingwConan(ConanFile):
+    name = "mingw"
+    description = "MinGW is a contraction of Minimalist GNU for Windows"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.mingw.org"
+    license = "ZPL-2.1", "MIT", "GPL-2.0-or-later"
+    topics = ("gcc", "gnu", "unix", "mingw32", "binutils")
+    settings = "os", "arch", "compiler"
+    options = {"threads": ["posix", "win32"], "exception": ["seh", "sjlj"]}
+    default_options = {"threads": "posix", "exception": "seh"}
+    build_requires = "7zip/19.00"
+    no_copy_source = True
+
+    def validate(self):
+        if self.settings.os != "Windows":
+            raise ConanInvalidConfiguration("MinGW is only supported by Windows.")
+        if self.settings.compiler == "gcc":
+            if str(self.settings.compiler.threads) != str(self.options.threads):
+                self.output.warn("MinGW threads may not differ from 'settings.compiler.threads'.")
+            if str(self.settings.compiler.exception) != str(self.options.exception):
+                self.output.warn("MinGW exception may not differ from 'settings.compiler.exception'.")
+        else:
+            self.output.warn("Only 'gcc' should be used as compiler.")
```

Comment:
Aren't we mixing build profile and host profile here?

This recipe will most often be used as a build requirement ==> used in the build context.
So the conan host profile should be checked for correct settings.
Looking at the [cross building documentatino](https://docs.conan.io/en/latest/systems_cross_building/cross_building.html), shouldn't `self.settings_target` be used?